src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -258,21 +259,11 @@ public ConnectionPool getNode(HostAndPort node) {
   }
 
   public ConnectionPool getSlotPool(int slot) {
-    r.lock();
```

Comment:
we could not remove this lock, otherwise, when reading, the other side is Array.full(null), which will read dirty data.